src/openrct2-ui/windows/Park.cpp

```diff
@@ -1495,6 +1511,199 @@ static void window_park_objective_paint(rct_window* w, rct_drawpixelinfo* dpi)
         dpi, ft.Data(), screenCoords, 221, ObjectiveNames[gScenarioObjective.Type], COLOUR_BLACK);
     screenCoords.y += 5;
 
+    // Your progress
+    // if (gScenarioObjective.Check() == ObjectiveStatus::Undecided)
+    if (gScenarioCompletedCompanyValue == MONEY32_UNDEFINED)
+    {
+        gfx_draw_string_left(dpi, STR_OBJECTIVE_PROGRESS_LABEL, nullptr, COLOUR_BLACK, screenCoords);
```

Comment:
I think the whole content of this `if` should be extracted to another function, something like `PaintProgressBar`. This is currently >300 lines added to a function that was already big.

Reply:
You're right. I'll split the window_park_objective_paint function in four separate functions. It'll be much better to read.

Reply:
> I think the whole content of this `if` should be extracted to another function, something like `PaintProgressBar`. This is currently >300 lines added to a function that was already big.

Phew... managed to split the function into four separate functions. I hope this worked out ok.

Reply:
Additionally, I've moved the check of the gScenarioCompletedCompanyValue into another function called GetCurrentObjectiveStatus(). This was a magical way of checking the objective status that also worked after the scenario was completet. Now the code is implemented in Scenario.cpp.